### DISCUSSION TITLE
Can you have reverse relations to abstract polymorphic base classes?

### DISCUSSION THREAD

**Discussion by incidentist** [jazzband/django-polymorphic#794] [Q&A] [answered]
I have a `Conversation` model that is made up of many `Interaction`s. `Interaction` is an abstract Polymorphic model with two concrete child models `Message` and `VoiceCall`. A conversation is all the messages and calls back and forth between two people. `Interaction` points back to `Conversation` like so:

```
conversation = models.ForeignKey(Conversation, on_delete=models.CASCADE, related_name='%(class)ss')
```

This creates properties `messages` and `voice_calls` on `Conversation` objects that give me all associated Interactions with those subclasses, but I also want a property `interactions` that gives me all Interactions associated with a `Conversation`, and it does not appear that Polymorphic models provide this on creation.

I understand why this is impossible with vanilla Django: I've told it that `Interaction` is abstract, so it should never create instances of it directly, and that's exactly what the reverse_relations would try to do. However, since Polymorphic knows how to figure out the subclass of each associated `Interaction`, it should be possible to get all `Interaction`s without violating the logic of abstract Models.

Is there a way to create this reverse relation on the abstract parent class?

**Comment by bckohan** [comment, marked as answer]
@incidentist  There is not a way to do what you are asking for because abstract models do not create database tables. You have two options

1. Make interaction non-abstract. This will create an interaction table and a child table for voicecall and message. When accessing interactions from conversation the polymorphic related manager will downcast each interaction to either a voicecall or message.
2. Leave interaction abstract and use [Django's GenericRelation feature](https://docs.djangoproject.com/en/6.0/ref/contrib/contenttypes/#generic-relations). The downside to generic relations is you lose the ability to do joins - so things like sorting across your interaction types will be difficult.
